core/unit_test/TestRange.hpp

```diff
@@ -332,19 +332,51 @@ TEST(TEST_CATEGORY, range_dynamic_policy) {
 #if !defined(KOKKOS_ENABLE_CUDA) && !defined(KOKKOS_ENABLE_HIP) && \
     !defined(KOKKOS_ENABLE_SYCL)
   {
-    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic> > f(0);
+    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic>> f(0);
     f.test_dynamic_policy();
   }
   {
-    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic> > f(3);
+    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic>> f(3);
     f.test_dynamic_policy();
   }
   {
-    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic> > f(1001);
+    TestRange<TEST_EXECSPACE, Kokkos::Schedule<Kokkos::Dynamic>> f(1001);
     f.test_dynamic_policy();
   }
 #endif
 }
 #endif
 
+void test_large_parallel_for_reduce() {
+  using ExecutionSpace              = typename TEST_EXECSPACE::execution_space;
+  constexpr long long unsigned size = 1llu << 32;
+  Kokkos::View<char *, TEST_EXECSPACE::memory_space> v(
```

Comment:
reducing it to a `char` view makes the test run in `defaultdevicetype.large_parallel_for_reduce (341 ms)` on MI300A. Pushing it into the `view_large_rank` and reusing the allocation adds saves ~50ms which I think is not worth having it in a weird place